### ISSUE TITLE
IllegalStateException: unmatched requested type Object on get("id", Object.class) and Integer-typed value

### ISSUE THREAD

**Issue by mp911de** [jasync-sql/jasync-sql#150]
When attempting to retrieve an `INT` value value via `Row.get(…)`, `JasyncRow.get(…)` fails with: 

```
java.lang.IllegalStateException: unmatched requested type Object
		at com.github.jasync.r2dbc.mysql.JasyncRow.get(JasyncRow.kt:41)
		at io.r2dbc.spi.Row.get(Row.java:101)
		at io.r2dbc.spi.Row.get(Row.java:127)
```

**SQL**

```sql
CREATE TABLE legoset (
    id          integer AUTO_INCREMENT PRIMARY KEY,
    name        varchar(255) NOT NULL,
    manual      integer NULL
) ENGINE=InnoDB;

INSERT INTO legoset (name, manual) VALUES(`Hello`, 0);
```

**Java Code**

```java
		Mono.from(connectionFactory.create())
				.flatMapMany(it -> it.createStatement("SELECT id FROM legoset").execute())
				.flatMap(it -> it.map((row, rowMetadata) -> row.get("id")))
				.as(StepVerifier::create)
				.expectNextCount(1)
				.verifyComplete();
```

R2DBC Version: 0.8.0.BUILD-SNAPSHOT

**Comment by oshai** [comment]
If I understand correctly when type is not specified (object) it is expected to return the value without conversion?

**Comment by mp911de** [comment]
Yes, that's the idea.

**Comment by oshai** [comment]
@mp911de - can you review #151 to verify?

**Comment by mp911de** [comment]
#151 looks good. Thanks for the prompt response.

**Comment by oshai** [comment]
fixed in 1.0.7
